File: test/python/test_write_events.py

import time
from sys import platform
import sys
sys.path.insert(0, '')

if platform == "linux" or platform == "linux2":
    import build.HarmonieReader as HarmonieReader
elif platform == "darwin":
    import build.HarmonieReader as HarmonieReader
elif platform == "win32":
    import build.Release.HarmonieReader as HarmonieReader

reader = HarmonieReader.HarmonieReader()
print("Test: test_write_events")

montage_index = 0
channel_name=""
filename = ""
if filename == "":
    print("ERROR: No filename specified")
    quit()

print(f"Opening file... :{filename}")
res = reader.open_file(filename)

# Adding a test group
group_name = "TestGroup"
group_desc = "TestGroup description"

print(f"Adding event group...")
print(f"  Group name:{group_name}")
print(f"  Group desc:{group_desc}")

group_index = reader.add_event_group(group_name, group_desc)

event_name = "test_write_events_name_" +  str(int(time.time()))
event_desc = "test_write_events_description"

start_time = 10
length = 1.5
number_of_events_to_create = 10

print(f"Adding {number_of_events_to_create} events  name:{event_name}")
channels = []
channels.append(channel_name)

for i in range(number_of_events_to_create):
    print(f"    event:{event_name} start:{start_time}")
    reader.add_event(event_name, group_name, start_time, length, channels, montage_index)
    start_time = start_time + 5

print("Saving file...")
reader.save_file()
print("Closing file...")
reader.close_file()

print("Checking if all events were created...")

validation_reader = HarmonieReader.HarmonieReader()
validation_reader.open_file(filename)

events = validation_reader.get_events()

number_of_events_created = 0
for e in events:
    if e.name == event_name:
        number_of_events_created = number_of_events_created + 1

if number_of_events_created == number_of_events_to_create:
    print("SUCCESS All events were properly created")
else:
    print("ERROR The new events were not all created")

validation_reader.close_file()
print("DONE")

File: test/cpp/test_write_events.cpp
// Your First C++ Program

#include <iostream>
#include <time.h>
#include "../src/harmonie_reader.h"

using namespace Harmonie;

int main() {
    HarmonieReader reader;
    std::cout << "Test: test_write_events" << std::endl;

    std::string filename = "";
    int montageIndex = 0;
    std::string channel = "C3-A2";
    if (filename == "") {
        std::cout << "ERROR: No filename specified" << std::endl;
        return 0;
    }

    std::cout << "Opening file... :" << filename << std::endl;
    bool res = reader.openFile(filename);

    if (!res) {
        std::cout << "ERROR Failed to open the file:" << filename << std::endl;
        return 0;
    } else {
        std::cout << "SUCCESS" << std::endl;
    }

    // Adding test group
    std::string groupName = "TestGroup";
    std::string groupDescription = "TestGroup Description";
    std::cout << "Adding event group... " << std::endl;
    std::cout << "  Group name: " << groupName << std::endl;
    std::cout << "  Description:" << groupDescription << std::endl << std::endl;

    // Create the test event
    //std::string eventName = "test_write_events_name_" + std::to_string(time(NULL));
    std::string eventName = "test_event";
    std::string eventDesc = "test_write_events_description";

    double startTime = 10;
    double timeLength = 5;
    int numberOfeventsToCreate = 10;
    std::cout << "Adding events..." << eventName << std::endl;

    std::vector<std::string> channels;
    channels.push_back(channel);

    for (int i=0; i < numberOfeventsToCreate; i++) {
        std::cout << "  event:\"" << eventName << "\" Start: " << startTime << " Length: " << timeLength <<std::endl;
        reader.addEvent(eventName, groupName, startTime, timeLength, channels, montageIndex);
        startTime += 5;
    }

    // Save file
    std::cout << "Saving file..." << std::endl;
    reader.saveFile();
    std::cout << "Closing file..." << std::endl;
    reader.closeFile();
    // Validate if the event has been created
    std::cout << "Checking if all events were created..." << std::endl;
    HarmonieReader validation_reader;
    res = validation_reader.openFile(filename);

    std::vector<CPSGFile::EVITEM> events = validation_reader.getEvents();
    
    int numberOfEventsCreated = 0;
    for (const auto& event: events) {
        if( event.Name == eventName) {
            numberOfEventsCreated++;
        }
    }
    
    if (numberOfeventsToCreate == numberOfEventsCreated) {
        std::cout << "SUCCESS All events were properly created" << std::endl;    
    } else {
        std::cout << "ERROR The new event was NOT found" << std::endl;    
    }
    
    validation_reader.closeFile();

    std::cout << "DONE" << std::endl << std::endl;
    return 0;
}
